uint8 game_progress
uint16 game_progress_remain
uint8 robot_id
uint16 sentry_hp
uint16 bullet_remain